Name: Hyperlinks in Descriptions
Version: 2.0.1-1
Author: Jason Robitaille
Description: Creates hyperlinks for text URLs in YouTube video descriptions  (MIT license)

--- .orig/usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
+++ /usr/palm/applications/com.palm.app.youtube/app/controllers/video-assistant.js
@@ -20,6 +20,24 @@
 		this.controller.setupWidget('share-menu', undefined, this.shareMenuModel);
 		this.controller.setupWidget(Mojo.Menu.commandMenu, {menuClass: "no-fade"}, this.cmdMenuModel);
 		
+		var s1 = this.video.desc.split("\n");
+		this.video.desc = "";
+		for(var i=0; i<s1.length; i++) {
+			var s2 = s1[i].split(" ");
+			for(var j=0; j<s2.length; j++) {
+				if(s2[j].indexOf("http://")==0 || s2[j].indexOf("www.")==0) {
+					this.video.desc += "<a href=\"" + s2[j] + "\">" + s2[j] + "</a>";
+				} else {
+					this.video.desc += s2[j];
+				}
+				if(j!=s2.length-1) {
+					this.video.desc += " ";
+				}
+			}
+			if(i!=s1.length-1) {
+				this.video.desc += "<br>";
+			}
+		}
 		// render video details template
         var info = Mojo.View.render({object: this.video, template: 'video/video-info'});
         $('video_info').update(info);
--- .orig/usr/palm/applications/com.palm.app.youtube/app/views/video/video-info.html
+++ /usr/palm/applications/com.palm.app.youtube/app/views/video/video-info.html
@@ -14,5 +14,5 @@
 	<div id="videoName" class="video-name">#{title}</div>
 
 	<div id="author" class="video-author">#{author}</div>
-	<div id="desc_preview" class="video-description">#{desc}</div>		
+	<div id="desc_preview" class="video-description">#{-desc}</div>
 </div>
\ No newline at end of file
